src/wireviz/wv_dataclasses.py

```diff
@@ -0,0 +1,815 @@
+    src: str
+    scale: Optional[ImageScale] = None
+    # Attributes of the image cell <td> containing the image:
+    width: Optional[int] = None
+    height: Optional[int] = None
+    fixedsize: Optional[bool] = None
+    bgcolor: SingleColor = None
+    # Contents of the text cell <td> just below the image cell:
+    caption: Optional[MultilineHypertext] = None
+    # See also HTML doc at https://graphviz.org/doc/info/shapes.html#html
+
+    def __post_init__(self):
+        self.bgcolor = SingleColor(self.bgcolor)
+
+        if self.fixedsize is None:
+            # Default True if any dimension specified unless self.scale also is specified.
+            self.fixedsize = (self.width or self.height) and self.scale is None
+
+        if self.scale is None:
+            if not self.width and not self.height:
+                self.scale = "false"
+            elif self.width and self.height:
+                self.scale = "both"
+            else:
+                self.scale = "true"  # When only one dimension is specified.
+
+        if self.fixedsize:
+            # If only one dimension is specified, compute the other
+            # because Graphviz requires both when fixedsize=True.
+            if self.height:
+                if not self.width:
+                    self.width = self.height * aspect_ratio(self.src)
+            else:
+                if self.width:
+                    self.height = self.width / aspect_ratio(self.src)
+
+
+@dataclass
+class PinClass:
+    index: int
+    id: str
+    label: str
+    color: MultiColor
+    parent: str  # designator of parent connector
+    _num_connections = 0  # incremented in Connector.connect()
+    _anonymous: bool = False  # true for pins on autogenerated connectors
+    _simple: bool = False  # true for simple connector
+
+    def __str__(self):
+        snippets = [  # use str() for each in case they are int or other non-str
+            str(self.parent) if not self._anonymous else "",
+            str(self.id) if not self._anonymous and not self._simple else "",
+            str(self.label) if self.label else "",
+        ]
+        return ":".join([snip for snip in snippets if snip != ""])
+
+
+@dataclass
+class Component:
+    category: Optional[str] = None  # currently only used by cables, to define bundles
+    type: Union[MultilineHypertext, List[MultilineHypertext]] = None
+    subtype: Union[MultilineHypertext, List[MultilineHypertext]] = None
+
+    # part number
+    partnumbers: PartNumberInfo = None  # filled by fill_partnumbers()
+    # the following are provided for user convenience and should not be accessed later.
+    # their contents are loaded into partnumbers during the child class __post_init__()
+    pn: str = None
+    manufacturer: str = None
+    mpn: str = None
+    supplier: str = None
+    spn: str = None
+    # BOM info
+    qty: NumberAndUnit = NumberAndUnit(1, None)
+    amount: Optional[NumberAndUnit] = None
+    sum_amounts_in_bom: bool = True
```

Comment:
It's not clear to me how to use `amount` and `sum_amounts_in_bom`. They are not mentioned in `syntax.md` and not used in any example/tutorial/test harness. I hope @formatc1702 can provide some examples.

Reply:
Sorry for the confusion. See my [comment in the `add_up_in_bom ` line](https://github.com/wireviz/WireViz/pull/251#discussion_r1359066031).